test/snapshot/__snapshots__/function.test.js.snap

```diff
@@ -189,6 +189,128 @@ Program {
 }
 `;
 
+exports[`Function tests first class callable support 1`] = `
+Program {
+  "children": Array [
+    ExpressionStatement {
+      "expression": Assign {
+        "kind": "assign",
+        "left": Variable {
+          "curly": false,
+          "kind": "variable",
+          "name": "callable",
+        },
+        "operator": "=",
+        "right": Call {
+          "arguments": Array [
+            229,
+          ],
+          "kind": "call",
+          "what": Name {
+            "kind": "name",
+            "name": "strlen",
+            "resolution": "uqn",
+          },
+        },
+      },
+      "kind": "expressionstatement",
+    },
+    ExpressionStatement {
+      "expression": Assign {
+        "kind": "assign",
+        "left": Variable {
+          "curly": false,
+          "kind": "variable",
+          "name": "callable",
+        },
+        "operator": "=",
+        "right": Call {
+          "arguments": Array [
+            229,
```

Comment:
this represents the `...` (ellipsis), right? The token codes are usually not part of our output - do you know how the official PHP parser represents this?

Reply:
Correct, that's the value of `T_ELLIPSIS`, fair point about the token codes not usually being used in the AST output. I did check how nikic/php-parser represents this, and it looks like this:

```
array(
    0: Stmt_Expression(
        expr: Expr_Assign(
            var: Expr_Variable(
                name: x
            )
            expr: Expr_FuncCall(
                name: Name(
                    parts: array(
                        0: test
                    )
                )
                args: array(
                    0: VariadicPlaceholder(
                    )
                )
            )
        )
    )
)
```

So then maybe we want a `VariadicPlaceholder` AST node. What do you think?

Reply:
Sounds good! :+1: